src-tauri/src/mod_management/mod.rs

```diff
@@ -130,3 +131,183 @@ pub fn get_installed_mods_and_properties(
+
+    // Early return on empty string
+    if thunderstore_mod_string.len() == 0 {
+        return Err("Passed empty string".to_string());
+    }
+
+    let deps = match get_mod_dependencies(thunderstore_mod_string.clone()).await {
+        Ok(deps) => deps,
+        Err(err) => return Err(err.to_string()),
+    };
+    dbg!(deps.clone());
+
+    // Recursively install dependencies
+    for dep in deps {
+        match fc_download_mod_and_install(game_install.clone(), dep).await {
+            Ok(()) => (),
+            Err(err) => {
+                if err.to_string() == "Cannot install Northstar as a mod!" {
+                    continue; // For Northstar as a dependency, we just skip it
+                }
+                else {
+                    return Err(err.to_string())
+                }
+            }
+        };
+    }
+
+    // Prevent installing Northstar as a mod
+    // While it would fail during install anyway, having explicit error message is nicer
+    let blacklisted_mods = ["northstar-Northstar", "northstar-NorthstarReleaseCandidate", "ebkr-r2modman"];
+    for blacklisted_mod in blacklisted_mods {
+        if thunderstore_mod_string.contains(blacklisted_mod) {
+            return Err("Cannot install Northstar as a mod!".to_string());
+        }
+    }
+
+    // Get download URL for the specified mod
+    let download_url = get_ns_mod_download_url(thunderstore_mod_string.clone()).await?;
+
+    // Create download directory
+    match std::fs::create_dir_all(download_directory.clone()) {
+        Ok(()) => (),
+        Err(err) => return Err(err.to_string()),
+    };
+
+    let name = thunderstore_mod_string.clone();
+    let path = format!(
+        "{}/___flightcore-temp-download-dir/{}.zip",
+        game_install.game_path, name
+    );
+
+    // Download the mod
+    let f = match thermite::core::actions::download_file(&download_url, path.clone()).await {
+        Ok(f) => f,
+        Err(e) => return Err(e.to_string()),
+    };
+
+    // Extract the mod to the mods directory
+    let pkg = match thermite::core::actions::install_mod(&f, std::path::Path::new(&mods_directory))
+    {
+        Ok(pkg) => pkg,
+        Err(err) => return Err(err.to_string()),
+    };
+    dbg!(pkg.clone());
+
+    // Add Thunderstore mod string to `mod.json` of installed NorthstarMods
+    for nsmod in pkg.mods {
+        let path_to_current_mod_json = format!(
+            "{}/{}/mod.json",
+            mods_directory,
+            nsmod.path.to_string_lossy()
+        );
+        match add_thunderstore_mod_string(path_to_current_mod_json, thunderstore_mod_string.clone())
+        {
+            Ok(()) => (),
+            Err(err) => {
+                println!("Failed setting modstring for {}", nsmod.name);
+                println!("{}", err);
+            }
+        }
+    }
+
+    // Delete downloaded zip file
+    std::fs::remove_file(path).unwrap();
+
+    // Delete temp download folder
```

Comment:
I suppose you can remove this before merge :)

Reply:
Oh yeah, removed with 29d1c644c39283f39a524b0517f3868150b58d1c o7